Support drawing graphs in the `test/examples` directory

skieffer opened the pull request:
I know that elkjs does not officially support the *drawing* of graphs (just layout), but I wondered if you would like to add a simple script that allows us to see the graphs in the `test/examples` directory.

I have found this very useful when working on ELK. My workflow is:

1. Make changes in ELK.
2. Rebuild elkjs in a sibling directory.
3. Reload a page from elkjs/test/examples in the browser.
4. Repeat.

I've written `SimpleGraphDrawer.js` which draws ELK graphs given in JSON format as SVG.
Furthermore, it handles all of the new JSON coordinate modes proposed in https://github.com/eclipse/elk/pull/1071 .

This PR includes four files:

* `SimpleGraphDrawer.js`: The drawing class (and a supporting `SvgDrawing` class).
* `test-browser-draw.html`: Same as the existing `test-browser.html`, except that now we draw the graph.
* `test-browser-draw2.html`: Like `test-browser-draw.html` but with a more interesting graph (namely, the one from [here](https://eclipse.dev/elk/documentation/tooldevelopers/graphdatastructure.html) ).
* `test-browser-draw-all-modes.html`: Draws a single graph 6 times, with all possible combinations of the new JSON coordinate modes.

skieffer commented:
I've added a new file, `test-coord-modes.html` which is designed to be a helpful demonstrator of the different coordinate modes, for new users. The JS within the file is (hopefully) well-commented, to help users see how to use the modes.

One purpose of including this could be to help head off issues like #332.

### Screenshots

There are some introductory notes:

![Screenshot 2025-04-06 at 08 30 53](https://github.com/user-attachments/assets/f8724a82-e289-4520-ab7a-f54d8a5a6862)

And here's the full table:

![Screenshot 2025-04-06 at 08 30 41](https://github.com/user-attachments/assets/8eefa4b3-2ac8-4808-9b91-a184fd542ea3)

skieffer commented:
I don' t know if there's any interest in merging this, but, if so, it might be worth discussing which of the five files it adds are really needed. Maybe they are a little redundant at this point.

soerendomroes reviewed (approved):
I am in favor of merging this. @Eddykasp any opinion on this?

Eddykasp commented:
I think this would be helpful :+1:

skieffer commented:
I made a few changes:

* Now showing the input graph on the `test-coord-modes.html` page.
  I think this is very helpful for users to understand what is being drawn.

* I renamed the class `SimpleGraphDrawer` --> `SimpleElkGraphDrawer`.
  This is a better (more specific) name, describing what the class actually does.

* I removed the file `test-browser-draw-all-modes.html` since I think it adds little
  value beyond what `test-coord-modes.html` already does.
